.ci/setup.sh

```diff
@@ -41,35 +39,30 @@ else  # Linux
             software-properties-common
 
         sudo apt-get install --no-install-recommends -y \
-            apt-utils \
             build-essential \
             ca-certificates \
             cmake \
             curl \
             git \
-            iputils-ping \
-            jq \
             libcurl4 \
             libicu-dev \
             libssl-dev \
-            libunwind8 \
             locales \
-            locales-all \
-            netcat \
-            unzip \
-            zip || exit 1
+            locales-all || exit 1
         if [[ $COMPILER == "clang" ]]; then
             sudo apt-get install --no-install-recommends -y \
                 clang \
                 libomp-dev
         elif [[ $COMPILER == "clang-17" ]]; then
-            sudo apt-get install wget
+            sudo apt-get install --no-install-recommends -y \
+                wget
             wget -qO- https://apt.llvm.org/llvm-snapshot.gpg.key | sudo tee /etc/apt/trusted.gpg.d/apt.llvm.org.asc
             sudo apt-add-repository deb http://apt.llvm.org/jammy/ llvm-toolchain-jammy-17 main
             sudo apt-add-repository deb-src http://apt.llvm.org/jammy/ llvm-toolchain-jammy-17 main
             sudo apt-get update
-            sudo apt-get install -y clang-17
-            sudo apt-get install --no-install-recommends -y libomp-17-dev
+            sudo apt-get install -y \
+                clang-17 \
+                libomp-17-dev
```

Comment:
With a `--no-install-recommends` added here, the C++ tests fail:

```text
[ 70%] Building CXX object CMakeFiles/lightgbm_objs.dir/src/treelearner/serial_tree_learner.cpp.o
In file included from /__w/1/s/src/treelearner/linear_tree_learner.cpp:7:
In file included from /__w/1/s/external_libs/eigen/Eigen/Dense:1:
In file included from /__w/1/s/external_libs/eigen/Eigen/Core:22:
In file included from /__w/1/s/external_libs/eigen/Eigen/src/Core/util/ConfigureVectorization.h:346:
In file included from /usr/lib/llvm-17/lib/clang/17/include/emmintrin.h:17:
In file included from /usr/lib/llvm-17/lib/clang/17/include/xmmintrin.h:31:
/usr/lib/llvm-17/lib/clang/17/include/mm_malloc.h:32:40: error: 'alloc_align' attribute parameter 1 is out of bounds
   32 |                                        __alloc_align__(2)))
```

https://dev.azure.com/lightgbm-ci/lightgbm-ci/_build/results?buildId=16401&view=results

I'm unsure why. For now, just letting `apt-get` install the other recommended packages seems to resolve that, so I think we should continue doing that.